src/codegen.c

```diff
@@ -411,22 +411,26 @@ static void codegen_expr(CodegenContext *ctx, Node *expr) {
 
     if (from_sz == to_sz) {
       ASSERT(expr->cast.value->ir, "May need to codegen cast value.");
-      expr->ir = expr->cast.value->ir;
+      expr->ir = ir_copy(ctx, expr->cast.value->ir);
+      expr->type = t_to;
+      ir_insert(ctx, expr->ir);
```

Comment:
I think you can just do 
```c
expr->ir = expr->cast.value->ir;
``` 
here, or am I missing something?

In any case, `expr->type` is already `t_to` (see line 397 :eyes:).

Reply:
Actually, no, what we should probably do is add an `ir_bitcast` instruction or sth that does nothing but ‘convert’ a value to a type w/ the same size. This is going to be necessary if you want that LLVM backend you mentioned.

Reply:
So on x86_64, it would simply end up becoming a copy, but other targets might have to do sth else

Reply:
Also, as an optimisation here, I suppose we could do:
```c
if (types_equal(expr->cast.value->type, expr->type)) expr->ir = expr->cast.value->ir;
else expr->ir = ir_bitcast(ctx, ir->type, expr->cast.value->ir);

Reply:
Interesting. I only added the copy so that the type information of the cast would be relayed to the IR. If the types are equal, *technically* we don't have to emit the copy but it will be yeeted anyways by RA or just not emitted due to being redundant.

Reply:
It seems like the `bitcast` instruction could take over here

Reply:
`bitcast` is a no-op, it's just a reinterpretation of the type, right?

Reply:
On x86_64, yes

Reply:
What might it have to do on other backends?

Reply:
Alright, I've implemented the bitcast change

https://github.com/LensPlaysGames/Intercept/pull/59/commits/0468cf7dcb863aa8c7c42822bc6f851ba423687a

Reply:
> What might it have to do on other backends? Out of curiosity

For instance, LLVM requires a bitcast sometimes, and there are separate instructions for pointer-to-integer and integer-to-pointer conversions.